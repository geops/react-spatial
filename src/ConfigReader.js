import TileLayer from 'ol/layer/Tile';
import XYZ from 'ol/source/XYZ';
import VectorSource from 'ol/source/Vector';
import WMTSSource from 'ol/source/WMTS';
import WMTSTileGrid from 'ol/tilegrid/WMTS';
import GeoJSONFormat from 'ol/format/GeoJSON';
import Layer from './Layer';
import VectorLayer from './VectorLayer';

import projections from './Projections';

class ConfigReader {
  static createXYZLayer(item) {
    const conf = { ...item };
    delete conf.data;

    return new Layer({
      ...conf,
      olLayer: new TileLayer({
        zIndex: -1,
        source: new XYZ({
          url: item.data.url,
        }),
      }),
    });
  }

  static createVectorLayer(item) {
    const conf = { ...item };
    delete conf.data;

    return new VectorLayer({
      ...conf,
      source: new VectorSource({
        url: item.data.url,
        format: new GeoJSONFormat(),
      }),
      style: item.data.style,
    });
  }

  static createWMTSLayer(item) {
    const conf = { ...item };
    delete conf.data;

    return new Layer({
      ...conf,
      olLayer: new TileLayer({
        zIndex: -1,
        source: new WMTSSource({
          url: item.data.url,
          tileGrid: new WMTSTileGrid({
            extent: item.data.projectionExtent,
            resolutions: item.data.resolutions,
            matrixIds: item.data.resolutions.map((res, i) => `${i}`),
          }),
        }),
      }),
    });
  }

  static createEmptyLayer(item) {
    const conf = { ...item };
    delete conf.data;

    return new Layer({
      ...conf,
    });
  }

  static createLayer(item) {
    let layer;

    switch (item.data.type) {
      case 'xyz':
        layer = ConfigReader.createXYZLayer(item);
        break;
      case 'wmts':
        layer = ConfigReader.createWMTSLayer(item);
        break;
      case 'vectorLayer':
        layer = ConfigReader.createVectorLayer(item);
        break;
      default:
        layer = ConfigReader.createEmptyLayer(item);
    }

    return layer;
  }

  constructor(map, data, style = null) {
    this.map = map;
    this.data = data;
    this.style = style;

    this.defaults = {
      data: {},
      visible: false,
      isBaseLayer: false,
    };
  }

  loadLayerFromConfig(config) {
    // apply default values
    const item = { ...this.defaults, ...config };

    if (item.data && item.data.type === 'wmts') {
      const proj = projections[item.data.projection];
      if (proj) {
        item.data.projectionExtent = proj.projectionExtent;
        item.data.resolutions = proj.resolutions;
      }
    }

    const layer = ConfigReader.createLayer(item);
    layer.init(this.map);

    if (item.children) {
      item.children.forEach(childConf => {
        layer.addChild(this.loadLayerFromConfig(childConf));
      });
    }

    return layer;
  }

  initialize() {
    return this.data.map(val => this.loadLayerFromConfig(val));
  }
}

const readConfig = (map, data) => {
  const configReader = new ConfigReader(map, data);
  return configReader.initialize();
};

const getVisibleTopic = topicList => {
  let visibleTopic = null;
  topicList.forEach(topic => {
    if (topic.visible) {
      visibleTopic = topic;
    }
  });
  return visibleTopic.children;
};

export default { getVisibleTopic, readConfig };
