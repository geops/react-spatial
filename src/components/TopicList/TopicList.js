import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LayerTree from '../LayerTree/LayerTree';
import Checkbox from '../Checkbox/Checkbox';
import Button from '../Button/Button';

const propTypes = {
  /**
   *  List of topics
   */
  topics: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      visible: PropTypes.bool,
      expanded: PropTypes.bool,
      children: PropTypes.array,
    }),
  ).isRequired,

  /**
   * All props aimed to be passed to child LayerTree
   */
  propsToLayerTree: PropTypes.object,

  /**
   * Function triggered on topic click.
   */
  onTopicClick: PropTypes.func,

  /**
   * Function triggered on topic toggle.
   */
  onTopicToggle: PropTypes.func,

  /**
   * CSS class to apply on the container.
   */
  className: PropTypes.string,

  /**
   * CSS class to apply on each item.
   */
  classNameItem: PropTypes.string,

  /**
   * CSS class to apply to the label element which contains the input.
   */
  classNameInput: PropTypes.string,

  /**
   * CSS class to apply to the toggle button which contains the title and the arrow.
   */
  classNameToggle: PropTypes.string,

  /**
   * CSS class to apply to the arrow.
   */
  classNameArrow: PropTypes.string,

  /**
   * Padding left.
   */
  padding: PropTypes.number,
};

const defaultProps = {
  propsToLayerTree: null,
  className: 'tm-topic-list',
  classNameItem: 'tm-topic-list-item',
  classNameInput: undefined,
  classNameToggle: 'tm-topic-list-toggle',
  classNameArrow: 'tm-topic-list-arrow',
  onTopicClick: () => {},
  onTopicToggle: () => {},
  padding: 10,
};

class TopicList extends Component {
  renderInput(topic) {
    const { onTopicClick, classNameInput } = this.props;

    return (
      <Checkbox
        inputType="radio"
        tabIndex={-1}
        checked={topic.visible}
        className={classNameInput}
        onClick={() => onTopicClick(topic)}
      />
    );
  }

  renderArrow(topic) {
    const { classNameArrow } = this.props;
    return (
      <div
        className={`${classNameArrow} ${classNameArrow}${
          topic.expanded ? '-expanded' : '-collapsed'
        }`}
      />
    );
  }

  renderBarrierFreeDiv(topic) {
    const { onTopicToggle, onTopicClick } = this.props;

    return (
      <div
        style={{
          position: 'absolute',
          margin: 'auto',
          top: 0,
          left: 0,
          bottom: 0,
          right: 0,
        }}
        role="button"
        tabIndex={0}
        onKeyPress={e => {
          if (e.which === 13) {
            if (topic.visible) {
              onTopicToggle(topic);
            } else {
              onTopicClick(topic);
            }
          }
        }}
      />
    );
  }

  // Render a button which expands/collapse the layer if there is children
  // or simulate a click on the input otherwise.
  renderToggleButton(topic) {
    const { onTopicToggle, onTopicClick, classNameToggle } = this.props;

    return (
      <Button
        tabIndex={-1}
        className={classNameToggle}
        onClick={() =>
          topic.visible ? onTopicToggle(topic) : onTopicClick(topic)
        }
      >
        <div>{topic.name}</div>
        {topic.visible ? this.renderArrow(topic) : null}
      </Button>
    );
  }

  renderTopic(topic, index) {
    const { classNameItem, padding, propsToLayerTree } = this.props;

    return (
      <div key={topic.id}>
        <div
          className={classNameItem}
          style={{
            paddingLeft: `${padding}px`,
          }}
        >
          {this.renderBarrierFreeDiv(topic)}
          {this.renderInput(topic)}
          {this.renderToggleButton(topic)}
        </div>
        <div style={{ paddingLeft: `${padding * 2}px` }}>
          {topic.visible && topic.expanded ? (
            <LayerTree key={index} {...propsToLayerTree} />
          ) : null}
        </div>
      </div>
    );
  }

  renderTree() {
    const { topics } = this.props;
    return <>{topics.map((t, i) => this.renderTopic(t, i))}</>;
  }

  render() {
    const { className, topics } = this.props;
    if (topics) {
      return <div className={className}>{this.renderTree()}</div>;
    }
    return null;
  }
}

TopicList.propTypes = propTypes;
TopicList.defaultProps = defaultProps;

export default TopicList;
