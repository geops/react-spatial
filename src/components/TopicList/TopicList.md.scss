.tm-topic-list-example {
  width: 100%;
  height: 400px;
  background: white;
  display: flex;
  overflow: hidden;
  border: 1px solid lightgray;

  .tm-map {
    width: auto;
    height: auto;
    flex-grow: 1;
  }

  .tm-topic-list-arrow {
    border: solid black;
    border-width: 0 1px 1px 0;
    display: inline-block;
    padding: 0;
    background: transparent;
    width: 5px;
    height: 5px;
  }

  .tm-topic-list-arrow-collapsed {
    transform: rotate(45deg);
    -webkit-transform: rotate(45deg);
    margin-top: -3px;
  }

  .tm-topic-list-arrow-expanded {
    transform: rotate(-135deg);
    -webkit-transform: rotate(-135deg);
    margin-top: 3px;
  }

  .tm-topic-list {
    width: 300px;
    padding: 10px;
    border-right: 1px solid lightgray;
    overflow-y: auto;
  }

  .tm-topic-list-item {
    display: flex;
    align-items: center;
    padding: 10px;
    position: relative;

    & > * {
      margin-right: 10px;
      align-items: center;
    }

    div {
      /* HACK: Without this it breaks the click on the label (Chrome) */
      position: relative;
    }

    /* CSS for the toggle div */
    .tm-topic-list-toggle {
      cursor: pointer;
      width: auto;
      height: auto;
      flex-grow: 1;
      display: flex;
      align-items: center;
      justify-content: left;

      div:first-child {
        margin-right: 10px;
      }
    }

    .tm-topic-list-input-radio span {
      border-radius: 50%;
    }

    /* Style the checkmark/indicator */
    .tm-topic-list-input-checkbox span::after {
      left: 5px;
      top: 1px;
      width: 5px;
      height: 10px;
      border: solid $brand-primary;
      border-width: 0 1px 1px 0;
      -webkit-transform: rotate(45deg);
      -ms-transform: rotate(45deg);
      transform: rotate(45deg);
    }

    /* Style the  (dot/circle) */
    .tm-topic-list-input-radio span::after {
      top: 2px;
      left: 2px;
      width: 12px;
      height: 12px;
      border-radius: 50%;
      background: $brand-primary;
    }
  }
}
