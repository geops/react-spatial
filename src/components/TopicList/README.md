#

This demonstrates the use of TopicList.

```jsx
import React from  'react';
import LayerTree from 'react-spatial/components/LayerTree';
import TopicList from 'react-spatial/components/TopicList';
import BasicMap from 'react-spatial/components/BasicMap';
import LayerService from 'react-spatial/LayerService';
import OLMap from 'ol/Map';
import ConfigReader from 'react-spatial/ConfigReader';

class TopicListExample extends React.Component {
  constructor(props) {
    super(props);

    this.center = [-10997148, 4569099];
    this.map = new OLMap({controls:[]});
    this.state = {
      topics: topicData,
      layerService: null,
    }
  }

  componentDidMount() {
    const layers = ConfigReader.readConfig(
      this.map,
      ConfigReader.getVisibleTopic(topicData),
    );

    this.setState({
      layerService: new LayerService(layers),
    })
  }

  onTopicClick(topic) {
    const { topics, layerService } = this.state;
    layerService.layers.forEach(layer => layer.setVisible(false));

    topics.map(t => {
      t.visible = (t.id === topic.id);
      t.expanded = true;
      return t;
    });

    const layers = ConfigReader.readConfig(
      this.map,
      ConfigReader.getVisibleTopic(topics),
    );

    this.setState({
      layerService: new LayerService(layers),
    });
  }

  onTopicToggle(topic) {
    const { topics } = this.state;

    this.setState({
      topics: topics.map(t => {
        if (t.id === topic.id) {
          t.expanded = !topic.expanded;
        }
        return t;
      })
    });
  }

  applyStyle(itemId) {
    this.layers[0].olLayer.getStyle().setText(
      new Text({
        font: '20px sans-serif',
        text: 'Last item modified is : ' + itemId,
      }),
    );
    this.layers[0].olLayer.changed();
  }

  render() {
    const {layerService, topics} = this.state;

    const propsToLayerTree = {
      layerService,
    };

    return (
      <div className="tm-topic-list-example">
        <TopicList
          onTopicClick={(topic) => this.onTopicClick(topic)}
          onTopicToggle={(topic) => this.onTopicToggle(topic)}
          propsToLayerTree={propsToLayerTree}
          layerService={layerService}
          topics={topics}
        />
        <BasicMap
          map={this.map}
          center={this.center}
          zoom={3}
          layers={this.layers}
        />
      </div>
    );
  }
}

<TopicListExample />;
```
