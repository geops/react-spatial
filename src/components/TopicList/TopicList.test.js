import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-canvas-mock';
import OLMap from 'ol/Map';
import renderer from 'react-test-renderer';
import TopicList from './TopicList';
import topicData from '../../../data/TopicData';
import ConfigReader from '../../ConfigReader';
import LayerService from '../../LayerService';

configure({ adapter: new Adapter() });

const mountTopicList = (topics, props) => {
  const layers = ConfigReader.readConfig(new OLMap({}), topics);
  const layerService = new LayerService(layers);
  return mount(
    <TopicList
      topics={topics}
      propsToLayerTree={{ layerService }}
      {...props || {}}
    />,
  );
};

const renderTopicList = (topics, props) => {
  const layers = ConfigReader.readConfig(
    new OLMap({}),
    ConfigReader.getVisibleTopic(topics),
  );
  const layerService = new LayerService(layers);
  const component = renderer.create(
    <TopicList
      topics={topics}
      propsToLayerTree={{ layerService }}
      {...props || {}}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
};

const toggleItem = '.tm-topic-list-toggle';

describe('Topiclist', () => {
  describe('matches snapshots', () => {
    test('using default properties.', () => {
      renderTopicList(topicData);
    });

    test('when no layers.', () => {
      const layerService = new LayerService();
      const component = renderer.create(
        <TopicList topics={topicData} propsToLayerTree={{ layerService }} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('when classNames are used.', () => {
      renderTopicList(topicData, {
        className: 'mimi',
        classNameItem: 'mama',
        classNameInput: 'momo',
        classNameToggle: 'mumu',
        classNameArrow: 'meme',
      });
    });
  });

  describe('triggers onTopicClick', () => {
    let wrapper;
    let spy;
    let spy2;
    const topicsData = [
      {
        id: 'topic2',
        name: 'Topic 2',
        visible: true,
        expanded: false,
        children: [],
      },
    ];

    const funcs = {
      topicClick: () => {},
      topicToggle: () => {},
    };

    const expectCalled = () => {
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy2).toHaveBeenCalledTimes(0);
    };

    beforeEach(() => {
      wrapper = mountTopicList(topicsData, {
        onTopicClick: () => funcs.topicClick(),
        onTopicToggle: () => funcs.topicToggle(),
      });
      spy = jest.spyOn(funcs, 'topicClick');
      spy2 = jest.spyOn(funcs, 'topicToggle');
    });

    afterEach(() => {
      spy.mockRestore();
      spy2.mockRestore();
    });

    test('when we press enter with keyboard on the label element.', () => {
      wrapper
        .find('label')
        .at(0)
        .simulate('keypress', { which: 13 });
      expectCalled();
    });

    test('when we click on input.', () => {
      wrapper
        .find('input')
        .at(0)
        .simulate('click');
      expectCalled();
    });
  });

  describe('triggers onTopicToggle', () => {
    let wrapper;
    let spy;
    const topicsData = [
      {
        id: 'topic2',
        name: 'Topic 2',
        visible: true,
        expanded: true,
        children: [],
      },
    ];

    const funcs = {
      topicToggle: () => {},
    };

    const expectCalled = () => {
      expect(spy).toHaveBeenCalledTimes(1);
    };

    beforeEach(() => {
      wrapper = mountTopicList(topicsData, {
        onTopicToggle: () => funcs.topicToggle(),
      });
      spy = jest.spyOn(funcs, 'topicToggle');
    });

    afterEach(() => {
      spy.mockRestore();
    });

    test('when we click on toggle button (label+arrow) of a topic', () => {
      wrapper
        .find(toggleItem)
        .first()
        .simulate('click');
      expectCalled();
    });
  });
});
