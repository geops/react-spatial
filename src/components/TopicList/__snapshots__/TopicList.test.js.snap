// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topiclist matches snapshots using default properties. 1`] = `
<div
  className="tm-topic-list"
>
  <div>
    <div
      className="tm-topic-list-item"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="tm-check tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={true}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="tm-topic-list-toggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 1
        </div>
        <div
          className="tm-topic-list-arrow tm-topic-list-arrow-expanded"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    >
      <div
        className="tm-layer-tree"
      >
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
              onKeyPress={[Function]}
              tabIndex={0}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="radio"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Vector sample layers
              </div>
              <div
                className="tm-layer-tree-arrow tm-layer-tree-arrow-expanded"
              />
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Points Samples
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Lines Samples
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Polygons Samples
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
              onKeyPress={[Function]}
              tabIndex={0}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Others layers
              </div>
              <div
                className="tm-layer-tree-arrow tm-layer-tree-arrow-expanded"
              />
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="checkbox"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Countries Borders
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="checkbox"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  USA Population Density
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
              onKeyPress={[Function]}
              tabIndex={-1}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                OSM Baselayer
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="tm-topic-list-item"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="tm-check tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={false}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="tm-topic-list-toggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 2
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    />
  </div>
</div>
`;

exports[`Topiclist matches snapshots when classNames are used. 1`] = `
<div
  className="mimi"
>
  <div>
    <div
      className="mama"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="momo tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={true}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="mumu"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 1
        </div>
        <div
          className="meme meme-expanded"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    >
      <div
        className="tm-layer-tree"
      >
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
              onKeyPress={[Function]}
              tabIndex={0}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="radio"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Vector sample layers
              </div>
              <div
                className="tm-layer-tree-arrow tm-layer-tree-arrow-expanded"
              />
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Points Samples
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Lines Samples
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-radio tm-radio"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="radio"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Polygons Samples
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
              onKeyPress={[Function]}
              tabIndex={0}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Others layers
              </div>
              <div
                className="tm-layer-tree-arrow tm-layer-tree-arrow-expanded"
              />
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="checkbox"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  Countries Borders
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="tm-layer-tree-item"
              style={
                Object {
                  "paddingLeft": "30px",
                }
              }
            >
              <label
                className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
                onKeyPress={[Function]}
                tabIndex={-1}
              >
                <input
                  checked={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  tabIndex={-1}
                  type="checkbox"
                />
                <span />
              </label>
              <div
                className="tm-layer-tree-toggle"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <div>
                  USA Population Density
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="tm-layer-tree-item"
            style={
              Object {
                "paddingLeft": "0px",
              }
            }
          >
            <label
              className="tm-layer-tree-input tm-layer-tree-input-checkbox tm-checkbox"
              onKeyPress={[Function]}
              tabIndex={-1}
            >
              <input
                checked={true}
                onChange={[Function]}
                onClick={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <span />
            </label>
            <div
              className="tm-layer-tree-toggle"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                OSM Baselayer
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="mama"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="momo tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={false}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="mumu"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 2
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    />
  </div>
</div>
`;

exports[`Topiclist matches snapshots when no layers. 1`] = `
<div
  className="tm-topic-list"
>
  <div>
    <div
      className="tm-topic-list-item"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="tm-check tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={true}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="tm-topic-list-toggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 1
        </div>
        <div
          className="tm-topic-list-arrow tm-topic-list-arrow-expanded"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    >
      <div
        className="tm-layer-tree"
      />
    </div>
  </div>
  <div>
    <div
      className="tm-topic-list-item"
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <div
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tabIndex={0}
      />
      <label
        className="tm-check tm-radio"
        onKeyPress={[Function]}
        tabIndex={-1}
      >
        <input
          checked={false}
          onChange={[Function]}
          onClick={[Function]}
          tabIndex={-1}
          type="radio"
        />
        <span />
      </label>
      <div
        className="tm-topic-list-toggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={-1}
      >
        <div>
          Topic 2
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20px",
        }
      }
    />
  </div>
</div>
`;
